core/lib/spree/permitted_attributes.rb

```diff
@@ -62,6 +64,11 @@ module PermittedAttributes
 
     @@line_item_attributes = [:id, :variant_id, :quantity]
 
+    @@menu_attributes = [:name, :locale, :location, :store_id]
+
+    @@menu_item_attributes = [:name, :subtite, :destination, :new_window, :item_type,
+                              :linked_resource_type, :linked_resource_id, :code, :menu_id]
```

Comment:
```suggestion
                              :linked_resource_type, :linked_resource_id, :code]
```

same here with menu id

Reply:
@damianlegawiec Can we nest `menu_items` under `menu` in the api `routes.rb` file for the platform API section? 

We need to associate a new menu_item with a menu on create, 

The following code neatly sets the new menu_item as a child of the menu.root, and so ideally we need an associated menu on create.

```ruby
# in core/app/models/menu_item.rb
    def ensure_item_belongs_to_root
      if menu.try(:root).present? && parent_id.nil?
        self.parent = menu.root

        store_new_parent
      end
    end
```

Reply:
@MatthewKennedy valid point, let's leave the menu_id here and do not nest the API edpoints